### PR TITLE
feat: payment reference utils

### PR DESCRIPTION
**Pull request by benjlevesque**

### CODE REVIEW

File: packages/payment-detection/src/utils.ts [hunk 1]
```diff
@@ -126,3 +127,47 @@ export const calculateEscrowState = (
   }
   return null;
 };
+
+/**
+ * Return the payment network extension of a Request.
+ */
+export function getPaymentNetworkExtension(
```
Comment (benjlevesque) [inline packages/payment-detection/src/utils.ts:134]: moved from packages/payment-processor/src/payment/utils.ts

File: packages/payment-detection/src/utils.ts [hunk 2]
```diff
@@ -126,3 +127,47 @@ export const calculateEscrowState = (
...
+  event: PaymentTypes.EVENTS_NAMES,
+) => {
+  if (event === PaymentTypes.EVENTS_NAMES.REFUND) {
+    return refundAddress || JSON.stringify(refundInfo);
+  }
+  return paymentAddress || JSON.stringify(paymentInfo);
+};
+
+/** Gets a payment (or refund) reference for any type of Request */
+export function getPaymentReference(
```
Comment (benjlevesque) [inline packages/payment-detection/src/utils.ts:157]: moved from packages/payment-processor/src/payment/utils.ts

File: packages/payment-detection/test/utils.test.ts [hunk 3]
```diff
@@ -160,3 +165,139 @@ describe('calculateEscrowState', () => {
     expect(calculateEscrowState(escrowEvents)).toEqual(PaymentTypes.ESCROW_STATE.PAID_ISSUER);
   });
 });
+
+fdescribe('getPaymentReference', () => {
```
Comment (AdamDowson) [inline packages/payment-detection/test/utils.test.ts:169]: Fdescribe

Comment (benjlevesque) [reply packages/payment-detection/test/utils.test.ts:169]: thanks!

Comment (benjlevesque) [reply packages/payment-detection/test/utils.test.ts:169]:
```suggestion
describe('getPaymentReference', () => {
```

**Review by AdamDowson — APPROVED** [review]

**Review by KolevDarko — APPROVED** [review]

File: packages/payment-processor/src/payment/utils.ts [hunk 4]
```diff
@@ -84,7 +72,7 @@ export function getPaymentNetworkExtension(
  */
 export function getRequestPaymentValues(request: ClientTypes.IRequestData): {
   paymentAddress: string;
-  paymentReference: string;
+  paymentReference: string | undefined;
```
Comment (olivier7delf) [inline packages/payment-processor/src/payment/utils.ts:75]: IMO:
paymentReference?: string;

Comment (benjlevesque) [reply packages/payment-processor/src/payment/utils.ts:75]: please use suggestion feature when making suggestions

File: packages/payment-detection/test/utils.test.ts [hunk 5]
```diff
@@ -160,3 +165,139 @@ describe('calculateEscrowState', () => {
     expect(calculateEscrowState(escrowEvents)).toEqual(PaymentTypes.ESCROW_STATE.PAID_ISSUER);
   });
 });
+
+describe('getPaymentReference', () => {
```
Comment (olivier7delf) [inline packages/payment-detection/test/utils.test.ts:169]: It doesn't make sense to test PaymentTypes.EVENTS_NAMES.ESCROW ?

File: packages/payment-processor/src/payment/utils.ts (hunk 4 above)
Comment (olivier7delf) [inline packages/payment-processor/src/payment/utils.ts:75]:
```suggestion
  paymentReference?: string;
```

**Review by olivier7delf — APPROVED** [review]

**Pull request merged**